uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java

```diff
@@ -464,7 +467,7 @@ private void queueCreationAndModificationEvent(final Set<Path> eventRealPaths,
                                 final WatchContext context = ((WatchContext) event.context());
                                 if (event.kind() == ENTRY_MODIFY || event.kind() == ENTRY_CREATE) {
                                     final Path path = context.getPath();
-                                    if (!path.getFileName().toString().startsWith(".")) {
+                                    if (!isDotFile(path)) {
                                         eventRealPaths.add(path);
```

Comment:
`eventRealPaths.add(path);` is not covered by test

Reply:
For nullable paths, I am returning true from isIgnored method.